### DISCUSSION TITLE
Introspection API

### DISCUSSION THREAD

**Discussion by pitermarx** [cake-build/cake#3366] [Q&A] [answered]
I was thinking that it might be useful to expose the information about the available tasks in the ICakeContext

The ICakeEngine has this `IReadOnlyList<ICakeTaskInfo> Tasks { get; }`

Does it make sense to make it available in the ICakeContext as well?

@patriksvensson

**Comment by pitermarx** [comment, marked as answer]
Actually, we can close this since the IScriptHost exposes `IReadOnlyList<ICakeTaskInfo>`
![image](https://user-images.githubusercontent.com/3081298/119118026-b9462080-ba21-11eb-8610-652a7ef6dfc8.png)

**Reply by pitermarx** [reply]
now I can do this
![image](https://user-images.githubusercontent.com/3081298/119125507-c1a25980-ba29-11eb-83dd-14cb63f196af.png)

**Reply by patriksvensson** [reply]
Doh, not sure how I missed that 😅
